### PR TITLE
fixes - #833

### PR DESCRIPTION
**Pull request by cjmctague**
Internal set used to track observed columns set to immutable sets/maps

### CODE REVIEW

File: modules/core/src/main/java/org/apache/fluo/core/observer/v2/ObserverStoreV2.java [hunk 1]
```diff
@@ -102,16 +104,16 @@ public RegisteredObservers load(CuratorFramework curator) throws Exception {
...
-    Set<Column> weakColumns = new HashSet<>();
-    Set<Column> strongColumns = new HashSet<>();
+    ImmutableSet.Builder<Column> weakColumnsBuilder = new ImmutableSet.Builder<Column>();
+    ImmutableSet.Builder<Column> strongColumnsBuilder = new ImmutableSet.Builder<Column>();
 
     for (Entry<Column, NotificationType> entry : jco.getObservedColumns().entrySet()) {
       switch (entry.getValue()) {
         case STRONG:
-          strongColumns.add(entry.getKey());
+          strongColumns = strongColumnsBuilder.add(entry.getKey()).build();
```
Comment (keith-turner) [inline modules/core/src/main/java/org/apache/fluo/core/observer/v2/ObserverStoreV2.java]: Building the map for each iteration of the loop may be expensive.  I would suggest only adding in the loop and then building the map after the loop.

File: modules/core/src/main/java/org/apache/fluo/core/observer/v1/ObserverStoreV1.java [hunk 2]
```diff
@@ -209,7 +212,7 @@ public Observers getObservers(Environment env) {
       }
 
       @Override
-      public Set<Column> getObservedColumns(NotificationType nt) {
+      public ImmutableSet<Column> getObservedColumns(NotificationType nt) {
```
Comment (keith-turner) [inline modules/core/src/main/java/org/apache/fluo/core/observer/v1/ObserverStoreV1.java]: Personally I would leave this type as `Set` and use `ImmutableSet` internally as the implementation.  However I can see that its useful to advertise that its immutable.

File: modules/core/src/main/java/org/apache/fluo/core/observer/ObserverUtil.java [hunk 3]
```diff
@@ -99,8 +97,8 @@ public String getObserverId(Column col) {
         }
 
         @Override
-        public Set<Column> getObservedColumns(NotificationType nt) {
-          return Collections.emptySet();
```
Comment (keith-turner) [inline modules/core/src/main/java/org/apache/fluo/core/observer/ObserverUtil.java]: I think `Collections.emptySet()` is immutable, so this change is not necessary.  However making the change does not hurt anything.

File: modules/core/src/main/java/org/apache/fluo/core/observer/v1/ObserverStoreV1.java [hunk 4]
```diff
@@ -161,7 +161,9 @@ private static void serializeObservers(DataOutputStream dos,
   private static Map<Column, ObserverSpecification> readObservers(DataInputStream dis)
       throws IOException {
 
-    HashMap<Column, ObserverSpecification> omap = new HashMap<>();
+    ImmutableMap.Builder<Column, ObserverSpecification> omapBuilder =
+        new ImmutableMap.Builder<Column, ObserverSpecification>();
+    ImmutableMap<Column, ObserverSpecification> omap = ImmutableMap.of();
```
Comment (keith-turner) [inline modules/core/src/main/java/org/apache/fluo/core/observer/v1/ObserverStoreV1.java]: It seems like `omap` could be dropped.

Comment (cjmctague) [inline modules/core/src/main/java/org/apache/fluo/core/observer/v1/ObserverStoreV1.java]: Just build in the return statement?

Comment (keith-turner) [inline modules/core/src/main/java/org/apache/fluo/core/observer/v1/ObserverStoreV1.java]: yeah, that would be a good way to get rid of it

**Comment by keith-turner** [comment]
@cjmctague would you like to try merging this to Apache git?  I'm on IRC if you have any questions.

**Comment by cjmctague** [comment]
@keith-turner I just dropped the omap. Yea I'll give that a go at someone point. Doing some office work  currently.

**Comment by keith-turner** [comment]
@cjmctague  the formatting in the latest commit is off.  Running `mvn clean compile` will fix it.

**Comment by cjmctague** [comment]
@keith-turner alright I'll fix that. Is [fluo-eclipse-style.xml](https://github.com/apache/incubator-fluo/blob/master/contrib/fluo-eclipse-style.xml) still correct? I used that to format the code.

**Comment by ctubbsii** [comment]
@cjmctague That style file may be out of date slightly... or it may work differently under different Eclipse versions. During the build, we use a maven plugin to format the code. That maven plugin uses a specific version of Eclipse libraries to do the job, so things are consistent. As long as you remember to do a build on the command-line and check in the changes, things should work out 95% of the time (checkstyle rules will probably catch the other 5%, requiring only slight manual tweaks).

**Comment by keith-turner** [comment]
> maven plugin uses a specific version of Eclipse libraries to do the job, so things are consistent.

I always use maven to format the code because of this.  One wrinkle though is that the maven build does not organize imports, that still needs to be done in the IDE.  I think some of the import changes in this PR fix a problem from a previous commit.

**Pull request merged**